app/src/main/java/org/systers/mentorship/viewmodels/HomeViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package org.systers.mentorship.viewmodels
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.util.Log
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.schedulers.Schedulers
+import org.systers.mentorship.MentorshipApplication
+import org.systers.mentorship.R
+import org.systers.mentorship.models.HomeStatistics
+import org.systers.mentorship.remote.datamanager.UserDataManager
+import org.systers.mentorship.utils.CommonUtils
+import org.systers.mentorship.utils.SingleLiveEvent
+import retrofit2.HttpException
+import java.io.IOException
+import java.util.concurrent.TimeoutException
+
+
+/**
+ * This class represents the ViewModel
+ * for the HomeFragment
+ */
+class HomeViewModel : ViewModel() {
+
+    private val TAG = this::class.java.simpleName
+    private val userDataManager by lazy { UserDataManager() }
+    private val compositeDisposable by lazy { CompositeDisposable() }
+
+    private val _userStats = MutableLiveData<HomeStatistics>()
+    private val _message = SingleLiveEvent<String>()
```

Comment:
Do you really need the SingleLiveEvent? Doesn't a fragment keep the configuration, and avoid the problems you mention as a reason to use SingleLiveEvent class? I used this in the past, but I can't' remember why I stopped using it.
cc @m-murad

Reply:
When `LifecycleOwner` is destroyed for recreation, `ViewModel` is preserved. Read [this](https://developer.android.com/topic/libraries/architecture/viewmodel)

Reply:
@m-murad I know this. Since the ViewModel is preserved, the LiveData objects we expose from it can hold on to their value. However, when the fragment is recreated and it re-subscribes to these live data objects, it get's updated with their latest values again. This is the expected behaviour in most cases, but when we are using LiveData for snackbars or toasts, then they also appear again on the screen. This is why SingleLiveEvent was created, right?